lib/order.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'menu'
+require_relative 'basket'
+
+class Order
+  attr_reader :basket, :menu
+  def initialize
+    @basket = Basket.new
+    @menu = Menu.new
+  end
+  def add(item, qty)
+    raise not_on_menu_error unless menu.has?(item)
+    basket.add(item, qty)
+  end
+  def remove(item, qty)
+    raise not_in_basket_error unless basket.has?(item)
+    basket.remove(item, qty)
+  end
+  def cancel
+    @basket  = Basket.new
+  end
+  def buy
+    print "#{confirmation_msg} #{delivery_time}\n\n"
+  end
+  def view
+    display_header
+    display_basket
+    display_total
+  end
+
+  private
+  def not_on_menu_error
+    "The requested item is not on the menu"
+  end
+  def not_in_basket_error
+    "You have not ordered this item"
+  end
+  def confirmation_msg
+    "Thank you. Your order was placed and will be delivered before"
+  end
+  def delivery_time
+    "#{(Time.now.hour + 1).modulo(24)}:#{Time.now.min}"
+  end
+  def display_header
+    print "\nYOUR ORDER\n\n"
+    printf("%-20s %5s %8s\n","ITEM","QTY","PRICE")
+  end
+  def display_basket
+    basket.each_item do |item, qty|
+      printf("%-20s %5d %8.2f \n", item, qty, menu.price(item))
```

Comment:
I've not seen `printf` used a lot in ruby — but I think this is a neat piece of code. I'd personally spend quite a while trying to figure out how to make it more readable, though. How could I make it clearer what's going on here?